inc/Engine/Optimization/Minify/CSS/AdminSubscriber.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace WP_Rocket\Engine\Optimization\Minify\CSS;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Minify/Combine CSS Admin subscriber
+ *
+ * @since 3.5.3
+ */
+class AdminSubscriber implements Subscriber_Interface {
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since 3.5.3
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		$slug = rocket_get_constant( 'WP_ROCKET_SLUG', 'wp_rocket_settings' );
+
+		return [
+			'update_option_' . $slug => [ 'clean_minify', 10, 2 ],
+		];
+	}
+
+	/**
+	 * Clean minify CSS files when options change.
+	 *
+	 * @since  3.5.3
+	 *
+	 * @param array $old_value An array of previous values for the settings.
+	 * @param array $value     An array of submitted values for the settings.
+	 */
+	public function clean_minify( $old_value, $value ) {
+		// Purge all minify cache files.
+		if ( ( isset( $old_value['minify_css'], $value['minify_css'] ) && $old_value['minify_css'] !== $value['minify_css'] )
+			||
+			( isset( $old_value['exclude_css'], $value['exclude_css'] ) && $old_value['exclude_css'] !== $value['exclude_css'] )
+			||
+			( isset( $old_value['cdn'], $value['cdn'] ) && $old_value['cdn'] !== $value['cdn'] )
+			||
+			( isset( $old_value['cdn_cnames'], $value['cdn_cnames'] ) && $old_value['cdn_cnames'] !== $value['cdn_cnames'] )
```

Comment:
I think we should use `array_key_exists()` instead of `isset()`. 

Why? `isset` returns `false` when (a) the key does not exist _or_ (b) the value for that element is `null`. 

In this code, _the `isset` serves to validate that the key exists_. Then the `!==` checks _if_ the old and new values are different.

Reply:
But what if: $old_value or $value are NULL. In this case with array_key_exists() a warning is displayed: `WARNING array_key_exists() expects parameter 2 to be array, null given` .
With array_key_exists() we need to check that also $value & $old_values are arrays.
I am pretty sure that under Integration tests these cases will occur.
So, do we really want to change the isset to array_key_exists?

Reply:
That's a valid point. What if either is not an array? You can add a guard clause to check _or_ expect that both must be an array by type hinting.

Reply:
I think a guard clause will do the trick in this case.
Will give it a try.